addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicThingHandler.java

```diff
@@ -105,9 +105,9 @@ public void run() {
                     // update configurations
                     Configuration config = editConfiguration();
                     for (HmChannel channel : device.getChannels()) {
+                        loadHomematicChannelValues(channel);
```

Comment:
The thing is, all datapoints (including VALUES) were fetched before as well (assuming all existing devices have at least one MASTER datapoint), but they were fetched _once per MASTER datapoint_. Taking the (random) example of HM-MOD-EM-8: a channel of it has 4 MASTER datapoints, so `loadHomematicChannelValues()` was run 4 times per channel before and is run one time per channel now. The only scenario where there would be more RPC calls than before is devices with channels that don't have any MASTER datapoints - are there any such devices? At least the 4 device types I own (HM-LC-Bl1-FM, HM-MOD-EM-8, HM-MOD-Re-8, HM-LC-Sw1-FM) have MASTER datapoints on all channels.

As you mentioned one could optimize further by only loading MASTER values instead of all values, but that's a larger/more involved change that I didn't feel like making at this point. I wanted to pick the low-hanging fruit first :-)

Reply:
You are right, this is something i need to optimize in the future. However, only the method was called four times. Due to the ```hmChannel.isInitialized()``` method, RPC call is done only once.

OK, then grab the low hanging fruits now ;-)